src/main/java/net/minecraftforge/event/world/MultiNoiseBiomeSourceLoadingEvent.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.world;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
+
+import com.mojang.datafixers.util.Pair;
+
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.biome.Climate.ParameterPoint;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.eventbus.api.EventPriority;
+
+/**
+ * Event for modifying multi-noise biome sources.
+ * This event is fired whenever a MultiNoiseBiomeSource is deserialized during dimension loading.
+ * This occurs after datapacks and worldgen registries load, but before chunk generators are constructed
+ * (and before the biome source itself is constructed).
+ * Standard tag wrappers are not resolvable at this time.
+ * 
+ * Best practice for use of this event:
+ * Additions during {@link EventPriority#HIGH}
+ * Removals during {@link EventPriority#NORMAL}
+ * Modifications during {@link EventPriority#LOW}
+ * 
+ * Cancelling this event will prevent the biome source from being modified.
+ * 
+ * This event is fired on the {@link MinecraftForge.EVENT_BUS}.
+ */
+@Cancelable
+public class MultiNoiseBiomeSourceLoadingEvent extends Event
+{
+    private final List<Pair<ParameterPoint, Supplier<Biome>>> parameters;
+    private final @Nullable ResourceLocation name;
```

Comment:
Current convention is that the nullability annotation for a field is specified on the line before:
```suggestion
    @Nullable
    private final ResourceLocation name;
```

Reply:
Fixed in next push